src/sentry/event_manager.py

```diff
@@ -674,29 +664,30 @@ def stringify(f):
             data['type'] = eventtypes.infer(data).key
             data['version'] = self.version
 
-        exception = data.get('exception')
-        stacktrace = data.get('stacktrace')
-        if exception and len(exception['values']) == 1 and stacktrace:
-            exception['values'][0]['stacktrace'] = stacktrace
-            del data['stacktrace']
+        exceptions = get_all_valid(data, 'exception', 'values')
+        stacktrace = get_valid(data, 'stacktrace')
+        if stacktrace and exceptions and len(exceptions) == 1:
+            exceptions[0]['stacktrace'] = stacktrace
```

Comment:
This code used to **remove** `event.stacktrace` and add it to the exception if it found an exception. The new code just copies the stacktrace into the exception but also keeps `event.stacktrace`. This breaks grouping.

afaict this is the only significant diff on kafka right now